ClientTests/ProfileTest.swift

```diff
@@ -12,4 +14,73 @@ class ProfileTest: XCTestCase {
     func withTestProfile(callback: (profile: Profile) -> Void) {
         callback(profile: MockProfile())
     }
+
+    func testShareItemAddsToDatabase() {
+        let url = "http://mozilla.com"
+        let title = "Mozilla"
+        let shareItem = ShareItem(url: url, title: title, favicon: nil)
+
+        // create clients
+        let client1GUID = Bytes.generateGUID()
+        let client2GUID = Bytes.generateGUID()
+
+        let clients = [RemoteClient(guid: client1GUID, name: "Test client 1", modified: NSDate.now(), type: "mobile", formfactor: "largetablet", os: "iOS"),
+            RemoteClient(guid: client2GUID, name: "Test client 2", modified: NSDate.now(), type: "desktop", formfactor: "laptop", os: "Darwin")]
+
+        class MockSQLiteCommands: SyncCommands {
+            // wipe all unsynced commands
+            func wipeCommands() -> Deferred<Result<()>> {
```

Comment:
`Success` is a `typealias` for `Deferred<Result<()>>`, and `succeed()` = `deferResult(())`.